torch/csrc/autograd/functions/convolution.cpp

```diff
@@ -807,13 +849,32 @@ static tensor_pair compute_grad_params(
             params.padding[1], params.padding[0],
             params.dilation[1], params.dilation[0], 1.0); goto done;
       } else {
-        /* CPU implementation has specialized MM kernels
-           for non-dilated case here */
-        at::SpatialConvolutionMM_accGradParameters(
-            input, grad_output, grad_weight, grad_bias, columns, ones,
-            kernel_size[1], kernel_size[0],
-            params.stride[1], params.stride[0],
-            params.padding[1], params.padding[0], 1.0); goto done;
+        if (params.use_nnpack(input)) {
+#ifdef WITH_NNPACK
+          // NNPACK does not have a bias gradient calculation, so we split
+          // into two calls here if necessary
+          nnpack::SpatialConvolution_accGradWeight(
+              input, grad_output, grad_weight,
+              kernel_size[1], kernel_size[0],
+              params.padding[1], params.padding[0]);
+
+          if (bias.defined() && params.should_compute_output(2)) {
+            // grad_output is in N, C, H, W, we re-shape and make contiguous
+            at::Tensor ones = grad_output.type().ones(input.size(0) * grad_output.size(2) * grad_output.size(3));
+            at::Tensor reshaped = grad_output.transpose(1, 3).contiguous().view({-1, ones.numel()});
+            grad_bias.addmv_(1.0, 1.0, reshaped, ones);
```

Comment:
is this really faster than a regular add with broadcasting?

Reply:
Do you mean making `ones` a Scalar?

Reply:
I mean `grad_bias = grad_output.sum(1)` or sth like this

Reply:
I will compare against transposing the channels to the innermost dim and doing a reduction as specified.

Reply:
In my test, the matrix multiply version was slightly faster or the same for a variety of different Conv params.

Reply:
Did you include the cost of allocating `ones` in here?

Reply:
Yes - it was testing the entire forward and backward passes, not just the particular function call.